Lib/test/test_builtin.py

```diff
@@ -321,7 +321,22 @@ def test_chr(self):
         self.assertRaises((OverflowError, ValueError), chr, 2**32)
 
     def test_cmp(self):
-        self.assertTrue(not hasattr(builtins, "cmp"))
+        self.assertTrue(hasattr(builtins, "cmp"))
+        self.assertEqual(cmp(-1, 1), -1)
+        self.assertEqual(cmp(1, -1), 1)
+        self.assertEqual(cmp(1, 1), 0)
+        # verify that circular objects are not handled
+        a = []; a.append(a)
+        b = []; b.append(b)
+        from UserList import UserList
+        c = UserList(); c.append(c)
```

Comment:
What is `UserList` for? Just a list that isn't the builtin `list`?

Reply:
Yes